js/bitso.js

```diff
@@ -1126,20 +1495,28 @@ module.exports = class bitso extends Exchange {
         //         }
         //     }
         //
-        currency = this.safeCurrency (undefined, currency);
+        const currencyId = this.safeString2 (transaction, 'currency', 'asset');
+        currency = this.safeCurrency (currencyId, currency);
+        const details = this.safeValue (transaction, 'details', {});
+        const datetime = this.safeString (transaction, 'created_at');
+        const withdrawalAddress = this.safeString (details, 'withdrawal_address');
+        const receivingAddress = this.safeString (details, 'receiving_address');
+        const networkId = this.safeString2 (transaction, 'network', 'method');
+        const status = this.safeString (transaction, 'status');
+        const withdrawId = this.safeString (transaction, 'wid');
         return {
-            'id': this.safeString (transaction, 'wid'),
-            'txid': undefined,
-            'timestamp': undefined,
-            'datetime': undefined,
-            'network': undefined,
-            'addressFrom': undefined,
-            'address': undefined,
-            'addressTo': undefined,
-            'amount': undefined,
-            'type': undefined,
-            'currency': currency['code'],
-            'status': undefined,
+            'id': this.safeString2 (transaction, 'wid', 'fid'),
+            'txid': this.safeString (details, 'tx_hash'),
+            'timestamp': this.parse8601 (datetime),
+            'datetime': datetime,
+            'network': this.safeNetwork (networkId),
+            'addressFrom': receivingAddress,
+            'address': withdrawalAddress !== undefined ? withdrawalAddress : receivingAddress,
```

Comment:
This line violates the following rule:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#using-ternary-conditionals

```suggestion
            'address': (withdrawalAddress !== undefined) ? withdrawalAddress : receivingAddress,
```